src/libYARP_robotinterface/src/yarp/robotinterface/Param.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include "Param.h"
+
+#include <yarp/os/LogStream.h>
+
+#include <yarp/os/Property.h>
+
+#include <string>
+
+
+std::ostream& std::operator<<(std::ostream &oss, const yarp::robotinterface::Param &t)
+{
+    oss << "(\"" << t.name() << "\"" << (t.isGroup() ? " [group]" : "") << " = \"" << t.value() << "\")";
+    return oss;
+}
+
+
+class yarp::robotinterface::Param::Private
+{
+public:
+    Private(Param * /*parent*/) :
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'Private' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=4763706)